test/acceptance/filepermissions_test.go

```diff
@@ -0,0 +1,118 @@
+	common.CopyFile(agentConfigLocalPath, agentConfigPath)
+	err := common.StartAgent(agentConfigPath, false)
+	if err != nil {
+		log.Printf("Agent failed to start due to err=%v\n", err)
+	}
+	time.Sleep(agentRuntime)
+	common.StopAgent()
+	groupResult := createGroupResult()
+	for _, testResult := range groupResult.TestResults {
+		assert.Equal(t, testResult.Status, status.SUCCESSFUL)
+	}
+	err = common.DeleteFile(agentConfigPath)
+	if err != nil {
+		log.Printf("Failed to delete config file; err=%v\n", err)
+	}
+}
+
+func createGroupResult() status.TestGroupResult {
+	testResults := make([]status.TestResult, 3)
+	testGroupResult := status.TestGroupResult{
+		Name:        "FilePermissions",
+		TestResults: testResults,
+	}
+	testResults[0] = onlyUserCanWriteToFilepath("root", agentConfigPath)
+	testResults[1] = onlyUserCanWriteToFilepath("cwagent", agentConfigCopiedDir)
+	testResults[2] = onlyUserCanWriteToFilepath("cwagent", translatedTomlPath)
+	return testGroupResult
+}
+
+func onlyUserCanWriteToFilepath(user, filepath string) status.TestResult {
+	log.Printf("validating only user=%v can write to filepath=%v", user, filepath)
+	testResult := status.TestResult{
+		Name:   filepath,
+		Status: status.FAILED,
+	}
+
+	hasExpectedOwner, err := fileHasOwnerAndGroup(filepath, user, user)
+	if err != nil || !hasExpectedOwner {
+		return testResult
+	}
+
+	hasExpectedPermission, err := fileHasPermission(filepath, filesystem.OwnerWrite, true)
+	if err != nil || !hasExpectedPermission {
+		return testResult
+	}
+
+	hasExpectedPermission, err = fileHasPermission(filepath, filesystem.AnyoneWrite, false)
+	if err != nil || !hasExpectedPermission {
+		return testResult
+	}
+	log.Printf("SUCCESS: only user=%v can write to filepath=%v", user, filepath)
+	testResult.Status = status.SUCCESSFUL
+	return testResult
+}
+
+func fileHasOwnerAndGroup(filepath, expectedOwner, expectedGroup string) (bool, error) {
+	owner, err := filesystem.GetFileOwnerUserName(filepath)
+	if err != nil {
+		err := fmt.Errorf("FAILED fileHasOwnerAndGroup(); error=%v", err)
+		log.Println(err)
+		return false, err
+	} else if owner != expectedOwner {
+		log.Printf("FAILED fileHasOwnerAndGroup(); expected OWNER=%v; actual=%v", expectedGroup, owner)
+		return false, nil
+	}
+
+	group, err := filesystem.GetFileGroupName(filepath)
+	if err != nil {
+		err := fmt.Errorf("FAILED fileHasOwnerAndGroup(); error=%v", err)
+		log.Printf("%v", err)
+		return false, err
+	} else if group != expectedGroup {
+		log.Printf("FAILED fileHasOwnerAndGroup(); expected GROUP=%v; actual=%v", expectedGroup, group)
+		return false, nil
+	}
+	return true, nil
+}
+
+func fileHasPermission(filepath string, permission filesystem.FilePermission, shouldHavePermission bool) (bool, error) {
+	hasPermission, err := filesystem.FileHasPermission(filepath, permission)
+	if err != nil {
+		return false, fmt.Errorf("fileHasPermission(), could not invoke filesystem.FileHasPermission; err=%v", err)
```

Comment:
nit: Looks like we log the error in the other functions. Would it make sense to log it here as well?

Reply:
good catch. I think I had some trouble thinking about when to log vs return an error